src/commands/commerce/extension/unmap.ts

```diff
@@ -0,0 +1,116 @@
+import { StatusFileManager } from '../../../lib/utils/statusFileManager';
+import { Result } from '../../../lib/utils/jsonUtils';
+
+Messages.importMessagesDirectory(__dirname);
+
+const TOPIC = 'extension';
+const CMD = `commerce:${TOPIC}:unmap`;
+const msgs = Messages.loadMessages('@salesforce/commerce', 'store');
+
+export class UnMapExtension extends SfdxCommand {
+    public static readonly requiresUsername = true;
+    public static description = msgs.getMessage('extension.unmap.cmdDescription');
+    public static example = [
+        `sfdx ${CMD} --registered-extension-name test-extension-name --store-name test-store-name `,
+        `sfdx ${CMD} --registered-extension-name test-extension-name --store-id test-store-id `,
+    ];
+    public static flagsConfig = {
+        'registered-extension-name': flags.string({
+            char: 'r',
+            description: msgs.getMessage('extension.unmap.regExtensionNameFlagDescription'),
+        }),
+        'store-name': flags.string({
+            char: 'n',
+            description: msgs.getMessage('extension.unmap.StoreNameFlagDescription'),
+        }),
+        'store-id': flags.string({
+            char: 'i',
+            description: msgs.getMessage('extension.unmap.storeId'),
+        }),
+    };
+    public org: Org;
+    public statusFileManager: StatusFileManager;
+
+    // eslint-disable-next-line @typescript-eslint/require-await
+    public async run(): Promise<void> {
+        this.unmapRecord(
+            this.flags['registered-extension-name'],
+            this.flags['store-name'],
+            this.flags['store-id'],
+            this.org.getUsername()
+        );
+    }
+
+    public unmapRecord(extensionName: string, storeName: string, storeId: string, userName: string): void {
+        if (storeName === undefined && storeId === undefined) {
+            throw new SfdxError(msgs.getMessage('extension.unmap.undefinedName'));
+        }
+        const storeid = this.validateStoreId(storeName, storeId, userName);
+        const name = forceDataSoql(`SELECT Name FROM WebStore WHERE Id='${storeid}' LIMIT 1`).result.records[0].Name;
+        try {
+            let deletedId: string;
+            const existingIds = forceDataSoql(
+                `SELECT DeveloperName,ExternalServiceProviderType FROM RegisteredExternalService WHERE DeveloperName='${extensionName}' AND ExternalServiceProviderType='Extension'`,
+                userName
+            );
+            for (const record of existingIds.result.records) {
+                const id = (record['ExternalServiceProviderType'] as string)
+                    .concat('__' as string)
+                    .concat(record['DeveloperName'] as string);
+                const deleteId = forceDataSoql(
+                    `SELECT Id FROM StoreIntegratedService WHERE Integration='${id}' AND StoreId='${storeid}'`,
+                    userName
+                ).result.records[0];
+                deletedId = deleteId.Id;
```

Comment:
currently when you try to unmap an extension that is not mapped (not in StoreIntegratedService table) it will return an error like this:
```
ERROR running commerce:extension:unmap:  Error while unmapping with the given registered extension name: JinsuTick
 Cannot read properties of undefined (reading 'Id')
```

I think it would be useful to display a more descriptive error message that the mapped extension with that registered extension name could not be found